prefect_aws/glue_job.py

```diff
@@ -0,0 +1,192 @@
+
+
+class GlueJobBlock(JobBlock):
+    """
+    Execute a job to the AWS Glue Job service.
+    Attributes:
+        job_name: The name of the job definition to use.
+        arguments: The job arguments associated with this run.
+            For this job run, they replace the default arguments set in the job
+            definition itself.
+            You can specify arguments here that your own job-execution script consumes,
+            as well as arguments that Glue itself consumes.
+            Job arguments may be logged. Do not pass plaintext secrets as arguments.
+            Retrieve secrets from a Glue Connection, Secrets Manager or other secret
+            management mechanism if you intend to keep them within the Job.
+            [doc](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html)
+        job_watch_poll_interval: The amount of time to wait between AWS API
+            calls while monitoring the state of a Glue Job.
+            default is 60s because of jobs that use AWS Glue versions 2.0 and later
+            have a 1-minute minimum.
+            [AWS Glue Pricing](https://aws.amazon.com/glue/pricing/?nc1=h_ls)
+    Example:
+        Start a job to AWS Glue Job.
+        ```python
+        from prefect import flow
+        from prefect_aws import AwsCredentials
+        from prefect_aws.glue_job import GlueJobBlock
+
+
+        @flow
+        def example_run_glue_job():
+            aws_credentials = AwsCredentials(
+                aws_access_key_id="your_access_key_id",
+                aws_secret_access_key="your_secret_access_key"
+            )
+            glue_job_run = GlueJobBlock(
+                job_name="your_glue_job_name",
+                arguments={"--YOUR_EXTRA_ARGUMENT": "YOUR_EXTRA_ARGUMENT_VALUE"},
+            ).trigger()
+
+            return glue_job_run.wait_for_completion()
+
+
+        example_run_glue_job()
+        ```"""
+
+    job_name: str = Field(
+        ...,
+        title="AWS Glue Job Name",
+        description="The name of the job definition to use.",
+    )
+
+    arguments: Optional[dict] = Field(
+        default=None,
+        title="AWS Glue Job Arguments",
+        description="The job arguments associated with this run.",
+    )
+    job_watch_poll_interval: float = Field(
+        default=60.0,
+        description=(
+            "The amount of time to wait between AWS API calls while monitoring the "
+            "state of an Glue Job."
+        ),
+    )
+
+    aws_credentials: AwsCredentials = Field(
+        title="AWS Credentials",
+        default_factory=AwsCredentials,
+        description="The AWS credentials to use to connect to Glue.",
+    )
+
+    async def trigger(self) -> GlueJobRun:
+        """trigger for GlueJobRun"""
+        return GlueJobRun(
+            job_name=self.job_name,
+            arguments=self.arguments,
+            job_watch_poll_interval=self.job_watch_poll_interval,
+        )
```

Comment:
I think we should start the job run here and then return the `GlueJobRun` object. The user can choose to wait for the job, or not, but the job should run whether or not they wait for it.

Reply:
Thank you for your comment.
I misunderstood the responsibilities of Glue Job Run.
It has been corrected.

https://github.com/PrefectHQ/prefect-aws/pull/308/commits/f3c7d2c2fecedbcf4673c8f444a1e8702060e7d9